src/sidebar/services/frame-sync.js

```diff
@@ -223,7 +225,7 @@ export class FrameSyncService {
     const addFrame = channel => {
       channel.call('getDocumentInfo', (err, info) => {
         if (err) {
-          channel.destroy();
+          channel.destroy(); // TODO: I am not sure if this works with `MessagePort`
```

Comment:
I think the idea here is that if the sidebar is unable to get the URL and metadata for a guest then it stops communicating with it and doesn't try to fetch annotations etc. I think that makes sense. For a `MessagePort` all we'd need to do is call `close()` on it.

Reply:
Thanks for the clarification.